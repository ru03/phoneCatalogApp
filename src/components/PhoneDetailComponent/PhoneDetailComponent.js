import React from 'react';
import classes from './PhoneDetailComponent.module.css';

const phoneDetailComponent = (props) => (
    <article className={classes.PhoneDetail}>
        <section>
            <img src={props.image} alt={props.model}></img>
        </section>
        <section>
            <h1>{props.brand}</h1>
            <div>
                <strong>Model: </strong>{props.model}
            </div>
            <div>
            <strong>Colors: </strong>{props.colors}
            </div>
            <div>
            <strong>Screen Type: </strong>{props.screen}
            </div>
            <div>
            <strong>Price: </strong>{props.price}
            </div>
        </section>
    </article>
)

export default phoneDetailComponent;