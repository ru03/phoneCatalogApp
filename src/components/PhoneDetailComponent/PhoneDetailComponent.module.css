.PhoneDetail {
    display: flex;
    justify-content: center;
    margin-top: 25px;
}

@media (max-width: 499px) {
    .PhoneDetail {
        margin: 25px;
        flex-direction: column;
    }
}