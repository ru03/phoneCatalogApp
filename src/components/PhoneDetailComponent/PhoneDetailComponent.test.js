import React from 'react';
import { mount } from 'enzyme';
import PhoneDetailComponent from './PhoneDetailComponent';

describe('PhoneDetail tests', () => {
    let phoneDetail;
    beforeEach(() => {
        phoneDetail = mount(
            <PhoneDetailComponent model='iphone8'
                price='15'
                colors='red'
                brand='apple'
                screen='screen'
                image='image'></PhoneDetailComponent>
        )
    })
    it('PhoneDetailComponent should be displayed', () => {
        expect(phoneDetail).toMatchSnapshot();
    });

    it('Props are correct', () => {
        expect(phoneDetail.props().model).toEqual('iphone8');
        expect(phoneDetail.props().price).toEqual('15');
        expect(phoneDetail.props().colors).toEqual('red');
        expect(phoneDetail.props().brand).toEqual('apple');
        expect(phoneDetail.props().screen).toEqual('screen');
        expect(phoneDetail.props().image).toEqual('image');
    });

    it('Should display brand in h1 tag', () => {
        expect(phoneDetail.find('#title').text()).toEqual('apple')
    })

})