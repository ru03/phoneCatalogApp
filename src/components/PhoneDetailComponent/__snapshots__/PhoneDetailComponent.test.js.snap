// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneDetail tests PhoneDetailComponent should be displayed 1`] = `
<phoneDetailComponent
  brand="apple"
  colors="red"
  image="image"
  model="iphone8"
  price="15"
  screen="screen"
>
  <article
    className="PhoneDetail"
  >
    <section>
      <img
        alt="iphone8"
        src="image"
      />
    </section>
    <section>
      <h1
        id="title"
      >
        apple
      </h1>
      <div>
        <strong>
          Model: 
        </strong>
        iphone8
      </div>
      <div>
        <strong>
          Colors: 
        </strong>
        red
      </div>
      <div>
        <strong>
          Screen Type: 
        </strong>
        screen
      </div>
      <div>
        <strong>
          Price: 
        </strong>
        15
      </div>
    </section>
  </article>
</phoneDetailComponent>
`;
